### ISSUE TITLE
build.rs exports and #[cfg(...)]

### ISSUE THREAD

**Issue by gaak99** [Wilfred/remacs#804]
Ran into a bug or at least a limitation in the generation of c_exports.rs by build.rs...

Example code
```
#[cfg(windows)]
#[no_mangle]
pub extern "C" fn file_attributes_internal(//snip) -> LispObject {
//snip
}
```
```make``` on linux
```
error[E0432]: unresolved import `dired::file_attributes_internal`
   --> /home/gb/warez/remacs-final/remacs/rust_src/target/release/build/remacs-3030dc6b627ee1fc/out/c_exports.rs:113:9
    |
113 | pub use dired::file_attributes_internal;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `file_attributes_internal` in `dired`. Did you mean to use `file_attributes_c_internal`?
```
Looks like the ```#[cfg(windows)]``` is not respected in build.rs (a quick look at the code confirms it).

Ditto for ```#[lisp_fn]```.

**Comment by gaak99** [comment]
Probably most of the time the exported fn will be on both windows/not(windows) but I have some code that is only for one of them.

**Comment by shaleh** [comment]
Good catch. Yep. I totally missed that. For now, can you add a dummy function?

**Comment by gaak99** [comment]
Giving it a try now...

BTW my first thought of a good quick fix would be if there is a rustc opt to expand a subset (cfg here) of the macros but the only related opt I could find expands all the macros
```
rustc -- -Z unstable-options --pretty=expanded
```
I then hacked up build.rs to filter my symbols which works fine but def just a temp workaround.

**Comment by gaak99** [comment]
Yea I'm able now to work around it (so no build.rs hacks needed to filter my symbols).
